### PR TITLE
Always draw hotkeys

### PR DESCRIPTION
**Pull request by vlolteanu**
This makes hotkeys always show up in the spell bar, as well as for the selected spell. It mimics DIablo 2.

We could also add an INI option for this.

### CODE REVIEW

**Comment by vlolteanu** [comment]
screen11

File: Source/control.cpp [hunk 1]
```diff
@@ -648,6 +648,21 @@ bool IsChatAvailable()
 #endif
 }
 
+namespace {
```
Comment (obligaron) [inline Source/control.cpp:651]: `control.cpp` already has a anonymous namespace.
Should we move `GetHotkeyName` inside of the existing one?

**Review by obligaron — APPROVED** [review]
lgtm.

My personal preference would be to not draw the hotkey in `DrawSpell` (active spell at buttom right), cause i don't think that it will help much. But you can discard that.

Another location where we could draw it would be the spell book.

File: Source/control.cpp [hunk 2]
```diff
@@ -747,13 +771,9 @@ void DrawSpellList(const Surface &out)
...
 		}
-		for (int t = 0; t < 4; t++) {
-			if (myPlayer._pSplHotKey[t] == spellId && myPlayer._pSplTHotKey[t] == spellListItem.type) {
-				auto hotkeyName = keymapper.KeyNameForAction(quickSpellActionIndexes[t]);
-				PrintSBookHotkey(out, spellListItem.location, hotkeyName);
-				strcpy(tempstr, fmt::format(_("Spell Hotkey {:s}"), hotkeyName.c_str()).c_str());
-				AddPanelString(tempstr);
-			}
+		if (hotkeyName) {
+			strcpy(tempstr, fmt::format(_("Spell Hotkey {:s}"), hotkeyName->c_str()).c_str());
```
Comment (glebm) [inline Source/control.cpp:775]:
```suggestion
			strcpy(tempstr, fmt::format(_("Spell Hotkey {:s}"), *hotkeyName).c_str());
```

File: Source/control.cpp (hunk 1 above)
Comment (vlolteanu) [reply Source/control.cpp:651]: You're right.

**Comment by glebm** [comment]
Conflicts are caused because moved the spell stuff out of `control.c` into separate files in https://github.com/diasurgical/devilutionX/pull/3585

**Review by AJenbo — APPROVED** [review]

**Comment by AJenbo** [comment]
Just noticed that the short cut is not displayed if the spell is disabled (firebolt in town)

**Comment by vlolteanu** [comment]
> Just noticed that the short cut is not displayed if the spell is disabled (firebolt in town)

Is the spell type set to `RSPLTYPE_INVALID` while in town?

**Comment by AJenbo** [comment]
idk

**Pull request merged**